lib/src.nix

```diff
@@ -14,12 +14,13 @@
   # name: name of the directory.
   #       Yes it is boring to always specify the name explicitly,
   #       but this way we guarantee no unexpected issues with caching.
-  # src: direcotry to clean. This is typically `./.`.
+  # src: directory to clean. This is typically `./.`.
   cleanGit = name: src:
     haskell-nix.lib.cleanGit {
-      src = haskell-nix.lib.cleanSourceWith {
-        inherit name src;
-      };
+      # For now, we'll use an unsafe operation
+      # FIXME: We probably want to update upstream.
+      src = builtins.unsafeDiscardStringContext
```

Comment:
I'm afraid this is not a solution. I suspect it casts the entire thing to string and thus breaks the composition. That's why I said that we need to discard something deep in haskell.nix.

Reply:
Check out `tests`, it checks that it at least somewhat works.

Reply:
I am not saying it doesn’t work, I am saying I am not sure it doesn’t break the composability that is at the core of this approach, and I think the tests don’t check this.

Reply:
Here is a test: add

```
foo = toString (subdir "test-foo" (cleanGit "test" ./.));
```

to your `default.nix`, evaluate this string. Then add a space to `default.nix`, evaluate the string again. These should be the same, because `foo` is supposed to depend only on the `test-foo` subdirectory, but they end up being different – the original top-level dir becomes an input of the expression. That is what I mean by “it breaks the composability” – it defeats the purpose of the `subDir` argument, you could just do `"${cleanGit ./.}/test-foo"` as well.